behave_django/environment.py

```diff
@@ -11,12 +11,12 @@ class PatchedContext(Context):
     def base_url(self):
         try:
             return self.test.live_server_url
-        except AttributeError:
+        except AttributeError as err:
             raise RuntimeError('Web browser automation is not available. '
                                'This scenario step can not be run with the '
-                               '--simple or -S flag.')
+                               '--simple or -S flag.') from err
 
-    def get_url(self, to=None, *args, **kwargs):
+    def get_url(self, *args, to=None, **kwargs):  # pylint: disable=invalid-name
```

Comment:
This change seems to cause the BDD tests to fail.

```python
@requires-live-http                                                                                                                                                                                                
Feature: URL helpers are available in behave's context # tests/acceptance/features/context-urlhelper.feature:2                                                                                                     
  In order to ensure that url helpers are available as documented                                                                                                                                                  
  As the Maintainer                                                                                                                                                                                                
  I want to test all suggested uses                                                                                                                                                                                
  Scenario: get_url() is an alias for the base_url property  # tests/acceptance/features/context-urlhelper.feature:7                                                                                               
    When I call get_url() without arguments                  # tests/acceptance/steps/context-urlhelper.py:6 0.000s                                                                                                
    Then it returns the value of base_url                    # tests/acceptance/steps/context-urlhelper.py:27 0.000s                                                                                               
                                                                                                                                                                                                                   
  Scenario: The first argument in get_url() is appended to base_url if it is a path  # tests/acceptance/features/context-urlhelper.feature:11                                                                      
    When I call get_url("/path/to/page/") with an absolute path                      # tests/acceptance/steps/context-urlhelper.py:11 0.000s                                                                       
    Then the result is the base_url with "/path/to/page/" appended                   # tests/acceptance/steps/context-urlhelper.py:32 0.000s                                                                       
      Assertion Failed: 'http://localhost:51171' != 'http://localhost:51171/path/to/page/'                                                                                                                         
      - http://localhost:51171                                                                                                                                                                                     
      + http://localhost:51171/path/to/page/                                                                                                                                                                       
      ?                       ++++++++++++++                                                                                                                                                                       
                                                                                                                                                                                                                   
                                                                                                                                                                                                                   
  Scenario: The reversed view path is appended to base_url if the first argument in get_url() is a view name  # tests/acceptance/features/context-urlhelper.feature:15                                             
    When I call get_url("admin:password_change") with a view name                                             # tests/acceptance/steps/context-urlhelper.py:16 0.000s                                              
    Then this returns the same result as get_url(reverse("admin:password_change"))                            # tests/acceptance/steps/context-urlhelper.py:51 0.005s                                              
    And the result is the base_url with reverse("admin:password_change") appended                             # tests/acceptance/steps/context-urlhelper.py:37 0.000s                                              
      Assertion Failed: 'http://localhost:43599' != 'http://localhost:43599/admin/password_change/'                                                                                                                
      - http://localhost:43599                                                                                                                                                                                     
      + http://localhost:43599/admin/password_change/                                                                                                                                                              
                                                                                                                                                                                                                   
                                                                                                                                                                                                                   
  Scenario: The model's absolute_url is appended to base_url if the first argument in get_url() is a model  # tests/acceptance/features/context-urlhelper.feature:20                                               
    When I call get_url(model) with a model instance                                                        # tests/acceptance/steps/context-urlhelper.py:21 0.000s                                                
    Then this returns the same result as get_url(model.get_absolute_url())                                  # tests/acceptance/steps/context-urlhelper.py:57 0.000s                                                
    And the result is the base_url with model.get_absolute_url() appended                                   # tests/acceptance/steps/context-urlhelper.py:44 0.000s                                                
      Assertion Failed: 'http://localhost:37467' != 'http://localhost:37467/behave/test/3/Foo'                                                                                                                     
      - http://localhost:37467                                                                                                                                                                                     
      + http://localhost:37467/behave/test/3/Foo                                                                                                                                                                   
```

```console
Failing scenarios:
  tests/acceptance/features/context-urlhelper.feature:11  The first argument in get_url() is appended to base_url if it is a path
  tests/acceptance/features/context-urlhelper.feature:15  The reversed view path is appended to base_url if the first argument in get_url() is a view name
  tests/acceptance/features/context-urlhelper.feature:20  The model's absolute_url is appended to base_url if the first argument in get_url() is a model
```

Reply:
@kingbuzzman I see that we really seem to have a deeper problem with the arguments of the `get_url` method.

The `to` argument should correspond to the first positional parameter of the Django [shortcut's `resolve_url` function](https://github.com/django/django/blob/main/django/shortcuts.py#L117-L129). Our implementation may break when `to` is something else than a string.

Specifically, when `get_absolute_url()` returns a URL containing a protocol and a hostname or IP address then there will be two `base_url`s in the resulting URL string. :lady_beetle: :bomb:

Reply:
The commit is 991672e571d2c653a6b794495fe71c67b7735f23. I force-pushed to undo the change.

I'll open a dedicated issue to address this problem separately.

Reply:
See #141 for a follow-up.